### PR TITLE
Add __pairs Symbol

### PR DESCRIPTION
**Pull request by daurnimator**
To allow customising `pairs` behaviour.

This can (should?) be extended with sensible __pairs for other standard JS classes.
  - Map
  - Array

Closes #1

### CODE REVIEW

File: src/jslib.js [hunk 1]
```diff
@@ -415,6 +415,54 @@ let jsmt = {
...
+	}
+};
+
+/* Create __pairs for all objects that inherit from Object */
+Object.prototype[Symbol.for("__pairs")] = function() {
+	return {
+		iter: function(last) {
+			let key = this.keys[this.index++];
+			if (!key)
+				return [];
```
Comment (daurnimator) [inline src/jslib.js]: Could return nothing here instead.

File: src/jslib.js [hunk 2]
```diff
@@ -415,6 +415,54 @@ let jsmt = {
...
+		return 3;
+	}
+};
+
+/* Create __pairs for all objects that inherit from Object */
+Object.prototype[Symbol.for("__pairs")] = function() {
+	return {
+		iter: function(last) {
+			let key = this.keys[this.index++];
+			if (!key)
```
Comment (daurnimator) [inline src/jslib.js]: Should be a check for `nil`, not just falsy

**Comment by daurnimator** [comment]
@giann do docs look okay? Also I wasn't super confident about setting the property on the global `Object.prototype`

**Comment by giann** [comment]
Yeah doc is ok. Since the property is a symbol i’m not too concerned about setting it on `Object.prototype`.

**Pull request merged**